### ISSUE TITLE
🐛 <Automaton equipment screen missing head, frame and attachments.>

### ISSUE THREAD

**Issue by Goliance** [DerpyProjectGroup/topaz#5555]
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x ] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x ] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)

Windows 11

## Branch affected by issue 

`base`

## Steps to reproduce

This happened after I received the sharpshot frame from Ghatsad, any previous attachments and the harlequin head and frame were gone and only recent attachments added show up. Logging in as puppetmaster shows nothing on the automaton equip screen. changing jobs does show the sharpshot frame and head but nothing else and only any attachments recently added. Making the character a GM and giving them the frames also did not work.

## Expected behavior

Head, Frame and attachments should all be the same before and after changing them or changing jobs.

![image](https://github.com/LandSandBoat/server/assets/105600004/520e7bc7-1e99-4bfc-b928-a3955f922ade)

**Comment by WinterSolstice8** [comment]
Will look into this tonight, no promises of a tonight fix though

**Comment by WinterSolstice8** [comment]
What level is your PUP?

**Comment by Goliance** [comment]
> What level is your PUP?

Level 12. I'm wondering if this happens every time you get a new frame?

And no worries, I'm just reporting it and if it gets fixed then thats great. Thanks for checking it out.

**Comment by WinterSolstice8** [comment]
couldn't reproduce, 
![image](https://github.com/LandSandBoat/server/assets/60417494/317e41e8-15f4-4ae2-a9b1-005799b7b102)

Can you get me a dump of the table `char_pet` row for your character id?

**Comment by Goliance** [comment]
Sorry Winter, I'm not familiar with how to do that, is it through heidi or via visual studio?

**Comment by WinterSolstice8** [comment]
Heidisql

**Comment by WinterSolstice8** [comment]
can you print the output of `/ver` in game? If you have any dat mods, does it still happen with them off?

Ideally, use a completely vanilla client with no mods/addons/plugins and see if you still have the problem

**Comment by Goliance** [comment]
[charpet.csv](https://github.com/LandSandBoat/server/files/15158461/charpet.csv)

Where do I get /ver? And I have no mods, running it straight off the server repo.

**Comment by WinterSolstice8** [comment]
It's a built in client command to FFXI

You are not running windower or ashita?

**Comment by Goliance** [comment]
Ah sorry I got it on windower, its ver30240327_1

I'll try it without any addons, etc now.

**Comment by WinterSolstice8** [comment]
So no .DAT mods of any kind? Just a vanilla install with windower, up to date, etc?

**Comment by Goliance** [comment]
Correct.

Edit: 
![image](https://github.com/LandSandBoat/server/assets/105600004/623a21eb-df94-48e5-a676-02c5473eab08)

No addons or plugins installed.

**Comment by WinterSolstice8** [comment]
Well, I certainly produced... something with your db values.

![image](https://github.com/LandSandBoat/server/assets/60417494/e0ea8a66-b99a-40d3-8d09-05ea11c8d130)

**Comment by Goliance** [comment]
Yep that looks about right.

**Comment by WinterSolstice8** [comment]
I had to manually query the db to insert the blobs and it looks better now:
![image](https://github.com/LandSandBoat/server/assets/60417494/251914c0-6f2d-403b-a33f-29c31de6b44f)

but it's still missing the original head/body, and not sure how many attachments. at least it looks real similar to your original error.

p.s. it seems LSB is not sending attachment equips until you open equip->master first, then equip->automaton will show (more) stuff

**Comment by Goliance** [comment]
I'm not sure how I got here, even the skill levels are back at 0 and there's a plethora of attachments missing. I'll create a new character and see if I can reproduce it.

The turning point though was right when I received the head+frame from the quest.

**Comment by WinterSolstice8** [comment]
![image](https://github.com/LandSandBoat/server/assets/60417494/9ac806e8-db9a-493e-b4b5-ce1c1646e2e6)
will give you back your frame/head that's missing

can you run the chat logger addon if you are going to remake and try to reproduce?

**Comment by Goliance** [comment]
Those commands reset the skill levels on the automaton but still no harlequin head+frame and no missing attachments.

**Comment by WinterSolstice8** [comment]
that's weird. They immediately came back for me... when was the last time you updated LSB?

**Comment by Goliance** [comment]
I updated it today actually.

**Comment by WinterSolstice8** [comment]
Alright, there was a period of time where blobs were written with some potentially wrong default values so I wonder if this was a side effect of it, but that entirely depends on when your character was created

**Comment by Goliance** [comment]
I'd say it was created last week. Now that I notice those commands give me an error, I'm supposed to run them inside the game correct?

**Comment by WinterSolstice8** [comment]
yeah you need to be a level 5 GM and you need to target yourself/the other player

**Comment by Goliance** [comment]
![image](https://github.com/LandSandBoat/server/assets/105600004/85ac5350-4a30-4a50-aec7-367c224ee3f0)

Frame command went through but not body and it still displays nothing.

![image](https://github.com/LandSandBoat/server/assets/105600004/bade38c5-5be2-4449-8fb0-1580b2f5adb8)

**Comment by WinterSolstice8** [comment]
Whoops, i cropped my screenshot wrong, 
![image](https://github.com/LandSandBoat/server/assets/60417494/776bd6ce-fac8-4819-a181-15fe4569586e)
its head and frame

**Comment by WinterSolstice8** [comment]
Once you run the commands, zone and then open the master equip then automaton equip, does it look normal then?

**Comment by Goliance** [comment]
Still nothing after zoning and opening the menus in that order, I also get this in the info bar;

![image](https://github.com/LandSandBoat/server/assets/105600004/d628e0af-e4be-4dc3-981d-83cfb8268f98)

**Comment by WinterSolstice8** [comment]
p.s. from your blob this is what I got for attachments:
![image](https://github.com/LandSandBoat/server/assets/60417494/af0ab165-37bf-409a-b03e-26e003f2c1b7)

**Comment by Goliance** [comment]
Those attachments are correct, I added them around the time I got the sharpshot and then all the other ones disappeared.

**Comment by WinterSolstice8** [comment]
paranoia check: try going outside to a non-city and seeing if the behavior changes?

**Comment by Goliance** [comment]
![image](https://github.com/LandSandBoat/server/assets/105600004/edcdd915-dd49-42e0-8587-e1286c3bcd1b)

I had to activate for the automaton equip screen to update but it shows 0 skills and no attachments. Progress? lol.

**Comment by WinterSolstice8** [comment]
Check the blob again in heidi, I wonder if it's zeroed on attachments

**Comment by Goliance** [comment]
Yep, shows NULL or a character that isn't recognized.

**Comment by WinterSolstice8** [comment]
can you upload it again? I know where to look sort of now at least...

**Comment by Goliance** [comment]
[charpet.csv](https://github.com/LandSandBoat/server/files/15158825/charpet.csv)

**Comment by WinterSolstice8** [comment]
yup. you do have less attachments now...  I don't really know what to think about this. I can't seem to reproduce the exact problem, BUT I am running on linux. Can you confirm it happens on a new char, too? If so I will boot up my VM and try to fix it maybe tomorrow. Assuming there's some obvious indicator of what breaks this

**Comment by Goliance** [comment]
Thats a copy, I'll get it done for you tomorrow, gonna have to call it a night here.

**Comment by WinterSolstice8** [comment]
I tried windows on my own just in case and couldn't replicate it. I am curious though... Is the server running on a non-english windows? Or see if this checkbox is checked on the server? (for now, make sure it is unchecked)

![image](https://github.com/LandSandBoat/server/assets/60417494/d6ce546a-669d-4907-ac09-db2061c08123)

**Comment by Goliance** [comment]
I wasn't able to reproduce the bug, which is great but that was weird...It was unchecked and its in english.

**Comment by WinterSolstice8** [comment]
No reproducing on a new character, right? That's suspicious.

Any chance you could upload the busted character's rows from the database so I can see if I can reproduce it locally? your char info doesnt include passwords or anything, so no worries.
